FRINXio/frinx-machine-e2e-tests
# fix failing tests

To prevent chrome to land on Aw, Snap! page
- https://stackoverflow.com/questions/51630143/cypress-long-automation-script-crashes-the-chrome-browser-with-aw-snap-error
  - Set "numTestsKeptInMemory": 0
  - Second, you can and should split your one large test file into multiple smaller test files

cypress.env.json
- Set "numTestsKeptInMemory": 0
- The number of tests for which snapshots and command data are kept in memory. Reduce this number if you are experiencing high memory consumption in your browser during a test run.

Split your one large test file into multiple smaller test files: ./2-devices:
    deviceTests2.cy.js
    deviceTests.cy.js
./4-workflow-manager:
    workflow_manager_tests2.cy.js
    workflow_manager_tests.cy.js

To refactor some tests which tended to fail during my testing to make them more robust:
- https://filiphric.com/8-common-mistakes-in-cypress-and-how-to-avoid-them
- 4-workflow-manager/workflow_manager_tests.cy.js in test 'filter by label' use intercept to wait until the data are delivered from the server
- add utils/graphql-test-utils.js to support intercepts of graphql requests https://docs.cypress.io/guides/end-to-end-testing/working-with-graphql#What-you-ll-learn
- 4-workflow-manager/workflow_manager_tests.cy.js in test 'install one device' use existing "data-cy" parameters as selectors which is more reliable in test 'try add wf to favourite'  use intercept to wait until the data are delivered from the server use intercept + wait instead explicit wait commands !!! in 'try
